### PR TITLE
Erase realm on new installation

### PR DESCRIPTION
**Pull request by janherich**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

We again introduced some breaking changes since last release (`0.9.18`), for example database encryption, so we need to go into "erase" mode again.
Since the top schema version in `0.9.18` is v2, just making sure that base/account realm schema versions are lower is enough, so this PR just compacts base schemas to v1 again (like it was done last time for the `0.9.17` release)

### Testing notes (optional):
Install this build on top of existing `0.9.18` build and observe that it works (by erasing the previous realm).

status: ready

### CODE REVIEW

**Comment by mandrigin** [comment]
@janherich do we want it to go to `develop` or `release/0.9.19` branch?

**Comment by janherich** [comment]
@mandrigin both

**Review by mandrigin — APPROVED** [review]

**Comment by flexsurfer** [comment]
@janherich do we want to delete realm data file and show the message?

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/UcCR5g for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/UcCR5g">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by yenda — APPROVED** [review]

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/3mou5w for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/3mou5w">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@janherich I installed iOS version on top of 0.9.18 (from TestFlight) and got the same error - "Realm file decription failed". I had 2 accounts, in 1 account there was public chat and some dapps.

I'm testing Android now

**Comment by lukaszfryc** [comment]
@janherich same on Android:

logcat
```
05-17 11:24:36.883 12629-12665/? D/JSRealm: Absolute path: /data/data/im.status.ethereum/files
05-17 11:24:40.229 12629-12724/? D/ReactNativeJS: DEBUG [status-im.utils.keychain:29] - Initializing realm encryption key...
05-17 11:24:40.463 12629-12724/? D/ReactNativeJS: DEBUG [status-im.data-store.realm.core:78] - Opening base realm... (first run)
05-17 11:24:40.468 12629-12724/? I/ReactNativeJS: PRETTY PRINTED EXCEPTION
    
    ***
    NAME: "Error"
    MESSAGE: Unable to open a realm at path '/data/data/im.status.ethereum/files/default.realm': Realm file decryption failed.
    
    schemaVersion@[native code]
    realm_version@index.android.bundle:12:5387847
    migrate_realm@index.android.bundle:12:5389638
    open_base_realm@index.android.bundle:12:5392967
    init@index.android.bundle:12:6677830
    G__36163_36165@index.android.bundle:12:6971311
    index.android.bundle:12:4002367
    invoke_interceptor_fn@index.android.bundle:12:3949733
    invoke_interceptors@index.android.bundle:12:3950034
    execute@index.android.bundle:12:3952404
    handle@index.android.bundle:12:3960896
    re_frame$router$IEventQueue$_process_1st_event_in_queue$arity$1@index.android.bundle:12:3995445
    re_frame$router$IEventQueue$_run_queue$arity$1@index.android.bundle:12:3977784
    index.android.bundle:12:3983536
    re_frame$router$IEventQueue$_fsm_trigger$arity$3@index.android.bundle:12:3987747
    index.android.bundle:12:3980302
    index.android.bundle:32:1369
    x@index.android.bundle:32:643
    callTimers@index.android.bundle:32:2788
    value@index.android.bundle:27:3582
    index.android.bundle:27:1067
    value@index.android.bundle:27:3009
    value@index.android.bundle:27:1039
    [native code]
    
    ***
```

File: env/dev/env/config.cljs [hunk 1]
```diff
@@ -2,5 +2,5 @@
 
 (def figwheel-urls {
                     :ios "ws://localhost:3449/figwheel-ws"
-                    :android "ws://10.0.2.2:3449/figwheel-ws"
-                    })
+                    :android "ws://10.0.3.2:3449/figwheel-ws"
```
Comment (rasom) [inline env/dev/env/config.cljs:5]: no need for this change in PR

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/RA2aaU for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/RA2aaU">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/data_store/realm/core.cljs [hunk 2]
```diff
@@ -22,9 +22,14 @@
...
   [file-name encryption-key]
-  (if encryption-key
-    (.schemaVersion rn-dependencies/realm file-name (to-buffer encryption-key))
-    (.schemaVersion rn-dependencies/realm file-name)))
+  ;; we need to try this if previous version was using unencrypted database
+  (or
+   (try
+     (.schemaVersion rn-dependencies/realm file-name)
+     (catch js/Object e
+       (do (log/info "Attempting to install over unencrypted database")
```
Comment (rasom) [inline src/status_im/data_store/realm/core.cljs]: `log/info` should return `nil`

**Review by rasom — APPROVED** [review]

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/vL91GK for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/vL91GK">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@janherich when I upgrade, the app looks fine (there are no old accounts) but when I create a new account an error is shown:

**Comment by lukaszfryc** [comment]
Android:

```
05-17 12:59:38.325 19483 19616 D ReactNativeJS: DEBUG [status-im.data-store.realm.core:36] - Opening realm at  6bd5559a5b21d8c893051439f0810de871045796 ...
05-17 12:59:38.335 19483 19616 D ReactNativeJS: DEBUG [status-im.data-store.realm.core:39] - Using encryption key...
05-17 12:59:38.351 19483 19616 I ReactNativeJS: PRETTY PRINTED EXCEPTION
05-17 12:59:38.351 19483 19616 I ReactNativeJS:
05-17 12:59:38.351 19483 19616 I ReactNativeJS: ***
05-17 12:59:38.351 19483 19616 I ReactNativeJS: NAME: "Error"
05-17 12:59:38.351 19483 19616 I ReactNativeJS: MESSAGE: Attempt to open unencrypted file with encryption key
05-17 12:59:38.351 19483 19616 I ReactNativeJS:
05-17 12:59:38.351 19483 19616 I ReactNativeJS: [native code]
05-17 12:59:38.351 19483 19616 I ReactNativeJS: open_realm@index.android.bundle:12:5389783
05-17 12:59:38.351 19483 19616 I ReactNativeJS: migrate_realm@index.android.bundle:12:5390766
05-17 12:59:38.351 19483 19616 I ReactNativeJS: open_migrated_realm@index.android.bundle:12:5391725
05-17 12:59:38.351 19483 19616 I ReactNativeJS: change_account@index.android.bundle:12:5399409
05-17 12:59:38.351 19483 19616 I ReactNativeJS: change_account@index.android.bundle:12:6679100
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:12:6685086
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:12:6685553
05-17 12:59:38.351 19483 19616 I ReactNativeJS: G__36175_36180@index.android.bundle:12:6974058
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:12:4002368
05-17 12:59:38.351 19483 19616 I ReactNativeJS: invoke_interceptor_fn@index.android.bundle:12:3949734
05-17 12:59:38.351 19483 19616 I ReactNativeJS: invoke_interceptors@index.android.bundle:12:3950035
05-17 12:59:38.351 19483 19616 I ReactNativeJS: execute@index.android.bundle:12:3952405
05-17 12:59:38.351 19483 19616 I ReactNativeJS: handle@index.android.bundle:12:3960897
05-17 12:59:38.351 19483 19616 I ReactNativeJS: re_frame$router$IEventQueue$_process_1st_event_in_queue$arity$1@index.android.bundle:12:3995446
05-17 12:59:38.351 19483 19616 I ReactNativeJS: re_frame$router$IEventQueue$_run_queue$arity$1@index.android.bundle:12:3977785
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:12:3983537
05-17 12:59:38.351 19483 19616 I ReactNativeJS: re_frame$router$IEventQueue$_fsm_trigger$arity$3@index.android.bundle:12:3987748
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:12:3980303
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:32:1369
05-17 12:59:38.351 19483 19616 I ReactNativeJS: x@index.android.bundle:32:643
05-17 12:59:38.351 19483 19616 I ReactNativeJS: callTimers@index.android.bundle:32:2788
05-17 12:59:38.351 19483 19616 I ReactNativeJS: value@index.android.bundle:27:3582
05-17 12:59:38.351 19483 19616 I ReactNativeJS: index.android.bundle:27:1067
05-17 12:59:38.351 19483 19616 I ReactNativeJS: value@index.android.bundle:27:3009
05-17 12:59:38.351 19483 19616 I ReactNativeJS: value@index.android.bundle:27:1039
05-17 12:59:38.351 19483 19616 I ReactNativeJS: [native code]
05-17 12:59:38.351 19483 19616 I ReactNativeJS:
05-17 12:59:38.351 19483 19616 I ReactNativeJS: ***
```

**Comment by flexsurfer** [comment]
please do not merge, i have PR with changes in schema too

**Comment by mandrigin** [comment]
@flexsurfer at least, we need that for the release branch :)

**Comment by flexsurfer** [comment]
@mandrigin it should be merged with this one https://github.com/status-im/status-react/pull/4321

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/eAHKzx for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/eAHKzx">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/x3Sks2 for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/x3Sks2">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/ohQwFA for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/ohQwFA">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/6MyknK for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/6MyknK">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/zTa14Z for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/zTa14Z">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/sS9MjV for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/sS9MjV">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@janherich works now 👍 

Should we merge it? Or wait for #4321 first (can be merged now).

**Comment by flexsurfer** [comment]
@janherich could you squash commits? and we can merge it, thanks

**Comment by janherich** [comment]
@flexsurfer done

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/Nd4BFn for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/Nd4BFn">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/fuan5Q for branch PR-4308

<blockquote><div><strong><a href="https://i.diawi.com/fuan5Q">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Pull request merged**